### PR TITLE
Tests to check for file creation

### PR DESCRIPTION
**Pull request by passy**

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -5,26 +6,76 @@ var assert  = require('assert');
...
-  before(helpers.before(path.join(__dirname, './temp')));
+  var webapp;
+
+  beforeEach(function (done) {
+    helpers.testDirectory(path.join(__dirname, 'temp'), function (err) {
+      if (err) {
+        done(err);
+      }
+
+      webapp = helpers.createGenerator('webapp:app', [
```
Comment (sindresorhus) [inline test/test.js]: this.webapp

File: test/test.js [hunk 2]
```diff
@@ -5,26 +6,76 @@ var assert  = require('assert');
 
 
 describe('Webapp generator test', function() {
-  before(helpers.before(path.join(__dirname, './temp')));
+  var webapp;
+
+  beforeEach(function (done) {
+    helpers.testDirectory(path.join(__dirname, 'temp'), function (err) {
+      if (err) {
+        done(err);
```
Comment (sindresorhus) [inline test/test.js]: return done(err);

File: test/test.js [hunk 3]
```diff
@@ -5,26 +6,76 @@ var assert  = require('assert');
...
 
   it('the generator can be required without throwing', function() {
     // not testing the actual run of generators yet
     this.app = require('../app');
   });
 
+  it('creates expected files in non-AMD mode', function (done) {
+    var expected= [
+      ['component.json', /"name":\s+"temp"/],
+      ['package.json', /"name":\s+"temp"/],
```
Comment (sindresorhus) [inline test/test.js]: `/"name": "temp"/`

**Comment by sindresorhus** [comment]
Except for the tiny nitpick, lgtm :)

**Comment by passy** [comment]
I :heart: code reviews. Updated. :)

File: test/test.js [hunk 4]
```diff
@@ -5,26 +6,73 @@ var assert  = require('assert');
...
+      done();
+    });
+  });
 
-  // FIXME
-  it.skip('creates expected files', function() {
+  it('creates expected files in AMD mode', function (done) {
+    var expected= [
+      ['component.json', /"name":\s+"temp"/],
+      ['package.json', /"name":\s+"temp"/],
```
Comment (sindresorhus) [inline test/test.js]: also meant this part, sorry

Comment (passy) [inline test/test.js]: Oops, of course.

**Pull request merged**